src/main/java/org/opentripplanner/standalone/config/framework/json/ParameterBuilder.java

```diff
@@ -80,6 +88,22 @@ public ParameterBuilder description(String description) {
     return this;
   }
 
+  /**
+   * Add documentation for optional field with default value to a parameter.
+   * <p>
+   * <b>Note!</b> This is only required if the default value in the documentation should be
+   * different from the value passed in with {@code #asNnnn(defaultValue)}. This may be the case if
+   * the value passed to {@code #asNnnn(defaultValue)} is derived from other parameters and the
+   * <em>code</em> default is just a fallback.
+   * <p>
+   * If the given default value is not of type String, then {@code String.valueOf(defaultValue)}
+   * is applied, there is
+   */
+  public ParameterBuilder docDefaultValue(Object defaultValue) {
```

Comment:
Do we need a separate method for this? Couldn't the existing methods with default values set this? I just fear that you either have to know this problem exists or spot that there is some difference in the document default value, and then look for this method. If the existing methods were modified, it would lessen the burden on the developer to spot this issue.

Reply:
It is not possible:

```
  .docDefaultValue(dft.sharedFilePattern().pattern())
  .asPattern(base.sharedFilePattern().pattern())
```          
(See [NetexConfig](src/main/java/org/opentripplanner/standalone/config/buildconfig/NetexConfig.java))
 
 You need to set two different values when the default used is calculated, and the only comes from the source defaults as a fallback. This only apply to explicit section where we have a default section as well. An alternative is to NOT document the default value in the explicit section - but we still need to communicate this in the code.